codebase2/codebase-sqlite/U/Codebase/Sqlite/Operations.hs

```diff
@@ -523,21 +521,25 @@ saveTermComponent h terms = do
             dependencies :: Set S.Reference =
               let (tmRefs, tpRefs, tmLinks, tpLinks) = TermUtil.dependencies tm
                   tpRefs' = Foldable.toList $ C.Type.dependencies tp
-                  getTermSRef :: S.Term.TermRef -> Maybe S.Reference
-                  getTermSRef (C.ReferenceBuiltin t) = Just (C.ReferenceBuiltin (tIds Vector.! fromIntegral t))
-                  getTermSRef (C.Reference.Derived (Just h) i) = Just (C.Reference.Derived (oIds Vector.! fromIntegral h) i)
-                  getTermSRef _selfCycleRef@(C.Reference.Derived Nothing _) = Nothing
+                  getTermSRef :: S.Term.TermRef -> S.Reference
+                  getTermSRef = \case
+                    C.ReferenceBuiltin t -> C.ReferenceBuiltin (tIds Vector.! fromIntegral t)
+                    C.Reference.Derived mh i ->
+                      C.Reference.Derived (maybe oId (\h -> oIds Vector.! fromIntegral h) mh) i
```

Comment:
Might actually still be worth it to split this into the `Nothing` and `Just` cases so you can add a comment on the Nothing case specifying that we include self-references in the dependency index.